reps/2023-10-13-accelerator-support.md

```diff
@@ -0,0 +1,234 @@
+
+### Should this change be within `ray` or outside?
+
+Inside `ray` project since we want to support those accelerators out of the box.
+
+## Stewardship
+
+### Required Reviewers
+The proposal will be open to the public, but please suggest a few experienced Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers.
+
+@pcmoritz, @ericl, @scv119
+
+### Shepherd of the Proposal (should be a senior committer)
+To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
+
+@scv119
+
+## Design
+
+### Ray Core
+
+#### API
+Users will use accelerators through Ray resources. For GPU accelerators (e.g. Nvidia, AMD, Intel GPUs), they will use the `GPU` resource name. For other accelerators, they will each have their own resource name (e.g. TPU accelerator has resource name `TPU`). `accelerator_type` option can also be used to specify a particular type of accelerator for a task or actor.
+
+```python
+@ray.remote(num_gpus=1)
+def task():
+    ...
+
+@ray.remote(num_gpus=1, accelerator_type="A100")
+def task():
+    ...
+
+@ray.remote(num_gpus=1, accelerator_type="Intel-Max-1550")
+def task():
+    ...
+
+@ray.remote(resources={"TPU": 1})
+def task():
+    ...
+
+@ray.remote(resources={"TPU": 1}, accelerator_type="TPU-V4")
+def task():
+    ...
+```
+
+##### Alternatives
+Other alternative APIs considered:
+
+Similar to `num_gpus`, having a top level parameter for each accelerator family:
+
+```python
+@ray.remote(num_tpus=1)
+def task():
+    ...
+
+@ray.remote(num_tpus=1, accelerator_type="TPU-V4")
+def task():
+    ...
+```
+
+A single `ACCELERATOR` resource for the ability to schedule a task on any accelerator
+and an `accelerator_family` is also introduced to allow specifying the family of the accelerators to use:
+
+```python
+@ray.remote(resources={"ACCELERATOR": 1})
+def task()
+    ...
+
+@ray.remote(resources={"ACCELERATOR": 1}, accelerator_family="TPU")
+def task()
+    ...
+
+@ray.remote(resources={"ACCELERATOR": 1}, accelerator_type="TPU-V4")
+def task()
+    ...
+```
+
+Similar to the above but having a top level `num_accelerators` parameter:
+
+```python
+@ray.remote(num_accelerators=1)
+def task()
+    ...
+
+@ray.remote(num_accelerators=1, accelerator_family="TPU")
+def task()
+    ...
+
+@ray.remote(num_accelerators=1, accelerator_type="TPU-V4")
+def task()
+    ...
+```
+
+#### Implementation
+
+Core defines an `Accelerator` base class and each accelerator needs to implement a subclass (e.g. `TPUAccelerator`):
+
+```python
+@DeveloperAPI
+class Accelerator(ABC):
+    @staticmethod
+    @abstractmethod
+    def is_available() -> bool:
```

Comment:
```suggestion
    def is_available_on_current_node() -> bool:
```

Otherwise, by default i'd assume this is checking if the accelerator is available on the cluster.